Description: Add support for clean python install mode
 Python build and install mechanism based on setup.py
 and pybuild.
 Permit a clean install through dh-python2
Author: Philippe Thierry
Last-Update: 2017-05-31
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,40 @@
+#!/usr/bin/python
+#
+# Authors:
+#   Philippe Thierry <phil@reseau-libre.net>
+
+import os
+import os.path
+
+from distutils.core import setup
+
+
+def get_packages():
+    # Distutils requires us to list all packages, this is very tedious and prone
+    # to errors. This method crawls the hierarchy and gathers all packages.
+
+    ret = ["backdoor_factory"]
+
+    for dirpath, _, files in os.walk("backdoor_factory"):
+        if "__init__.py" in files:
+            ret.append(dirpath.replace(os.path.sep, "."))
+
+    return ret
+
+
+setup(
+    name="backdoor_factory",
+    #version=version.VERSION_STRING,
+    author="Philippe Thierry",
+    author_email="phil@reseau-libre.net",
+    description="...",
+    license="LGPL2.1+",
+    url="https://github.com/secretsquirrel/the-backdoor-factory",
+    packages=['intel','arm','winapi','onionduke','preprocessor'],
+    py_modules=['pebin','elfbin','machobin'],
+    scripts=[
+        os.path.join("./", "backdoor.py"),
+    ],
+    #data_files=[
+    #],
+)
